src/Integration.Vsix/CFamily/VcxProject/CmdBuilder.cs

```diff
@@ -0,0 +1,394 @@
+﻿/*
+ * SonarLint for Visual Studio
+ * Copyright (C) 2016-2022 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using System;
+using System.Text;
+using System.Text.RegularExpressions;
+using Microsoft.VisualStudio;
+using Microsoft.VisualStudio.VCProjectEngine;
+
+namespace SonarLint.VisualStudio.Integration.Vsix.CFamily.VcxProject
+{
+    internal class CmdBuilder
```

Comment:
No. This file is loading options from the UI and is specific to VS. the idea was that CFamily java side change on every release. It is really hard to synchronize especially after a lot of refactoring on the java side that has no impact on SonarLint.  We had a lot of bugs because it is hard to synchronize. Now we have only have this code that is specific to the CFamily compilation options. It is localized and isolated. The idea is to be able to update the CFamily analyzer without any changes on the SonarLint side. Like the update to 6.30, 6.31..
Later, we can even drop the entire file if we find a way to load All option UI (as mentioned in #2661)

Of course `CmdBuilder.cs` need to be maintained to handle two things: discovering bugs and VS adding new compiler options in their UI(which doesn't happen that often). I'm happy to maintain `CmdBuilder.cs` which should be low effort. It should be the only place that contains CFamily specific code that supposes to evolve with new.